test/jdk/javax/swing/JSplitPane/TestSplitPaneEnableTest.java

```diff
@@ -0,0 +1,150 @@
+            if (laf.getClassName().toLowerCase().contains("gtk")) {
+                continue;
+            }
+            System.out.println("Testing LAF : " + laf.getClassName());
+            SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
+
+            try {
+                SwingUtilities.invokeAndWait(() -> {
+                    frame = new JFrame("SplitPaneTest");
+                    jsp = new JSplitPane(JSplitPane.HORIZONTAL_SPLIT,
+                                         new JButton("Left"),
+                                         new JButton("Right"));
+
+                    frame.add(jsp);
+                    jsp.setUI(new TestSplitPaneUI());
+                    jsp.setOneTouchExpandable(true);
+
+                    jsp.setEnabled(false);
+
+                    frame.setSize(400, 200);
+                    frame.setLocationRelativeTo(null);
+                    frame.setVisible(true);
+                });
+
+                robot.waitForIdle();
+                robot.delay(1000);
+
+                SwingUtilities.invokeAndWait(() -> {
+                    loc = leftOneTouchButton.getLocationOnScreen();
+                    prevDivLoc = jsp.getDividerLocation();
+                });
+                robot.mouseMove(loc.x, loc.y);
+                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+                robot.waitForIdle();
+                robot.delay(1000);
+
+                SwingUtilities.invokeAndWait(() -> {
+                    curDivLoc = jsp.getDividerLocation();
+                });
+                System.out.println("current Divider location " + curDivLoc);
+                if (curDivLoc != prevDivLoc) {
+                    throw new RuntimeException("Divider position changed on disabled oneTouchExpandable arrow click");
+                }
+            } finally {
+                SwingUtilities.invokeAndWait(() -> {
+                    if (frame != null) {
+                        frame.dispose();
+                    }
+                });
+            }
+        }
+    }
+
+    static class TestSplitPaneUI extends BasicSplitPaneUI {
+
+        public TestSplitPaneUI() {
+            super();
+        }
+
+        public BasicSplitPaneDivider createDefaultDivider() {
+            return new TestSplitPaneDivider(this);
+        }
+    }
+
+    static class TestSplitPaneDivider extends BasicSplitPaneDivider {
+
+        public TestSplitPaneDivider(BasicSplitPaneUI ui) {
+            super(ui);
+        }
+
+        protected JButton createLeftOneTouchButton() {
+            leftOneTouchButton = super.createLeftOneTouchButton();
+            return leftOneTouchButton;
+        }
+
+        protected JButton createRightOneTouchButton() {
+            rightOneTouchButton = super.createRightOneTouchButton();
+            return rightOneTouchButton;
+        }
```

Comment:
Alternatively, you could add methods which return `leftButton` and `rightButton` fields.